client/src/app/project-registration/services/autofill-project.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {map} from 'rxjs/operators';
+import {Observable, throwError} from 'rxjs';
+import {EuropePMCHttpSearchResponse, Identifier} from '../models/europepmcsearch';
+import {AutofillProject} from '../models/autofill-project';
+
+@Injectable()
+export class AutofillProjectService {
+  API_URL = 'https://www.ebi.ac.uk/europepmc/webservices/rest/search';
+  DOI_BASE_URL = 'https://doi.org/';
+
+  constructor(private http: HttpClient) {}
+
+  getProjectDetails(searchUsing: Identifier, searchString: string): Observable<AutofillProject> {
+    return this.queryEuropePMC(searchUsing, searchString).pipe(
+      map(response => {
+        const result = response.resultList.result[0];
+        if (result) {
+          const projectDetails: AutofillProject = {
```

Comment:
Is there test code for all the conversions that is going on here?

Reply:
not yet, no